### ISSUE TITLE
Move databases from Modulus to Compose + general infrastructure

### ISSUE THREAD

**Issue by drenfr01** [drenfr01/unionCapital#267]
Modulus is beginning to make me nervous about their DB hosting for the following 4 reasons:

1) They *still* haven't really answered my support ticket from 3 weeks ago, despite me asking for updates every week ("we're working on it" has been their only response). This means that we aren't being billed for storage (which is good...unless we have an expensive delayed bill coming...), but also have 0 metrics on how our database is performing. 

2) Modulus doesn't do daily backups. They have "disaster recovery" backups every 12 hours, but it's not clear we can use them. So right now before each deploy I manually do backups, but this has historically been ~1 week apart. 

3) Modulus doesn't do oplog tailing. This, according to both Kadira and Meteor (http://info.meteor.com/blog/david-glasser-on-scaling-meteor-with-the-mongodb-oplog) has huge performance benefits, ~2.5x - 10x. 

Given all the following, here's what I propose our new infrastructure looks like:

1) Move databases to Compose. This gives us daily backups for free (up to 7 rolling days), great metrics, and oplog tailing. The cost here is $18/GB, which is a bit pricey given that we have 4 GB. However, the vast majority (like 3.5+ GB) lives in images, which leads me too:

2) Store images in Amazon S3 instead of in the DB. This *might* improve performance, but most importantly this makes storage super cheap so we can always keep images and pay like a crazy low rate per GB (I saw on Amazon's web site like $.00003 per GB or something...)

Bottom line: slight increase in price, probably on the order of $10 a month (although we might be able to decrease our cost by moving back down to 1 servo on Modulus, because I'm not sure if it's really helping performance...) Slight increase in complexity of our infrastructure, because we're now using 3 services. But...more confidence, better performance, and probably a more stable platform to scale. 

@TylerSheffels @drewproud @ballek @unioncapitalboston Thoughts?

**Comment by drewproud** [comment]
That's some great research, Duncan. Not having accessible daily backups is stressing me out. 

It's also a great move with the future in mind - we're going to have to address this sooner or later as our userbase continues to grow.

Thumbs up.

**Comment by gpatkinson** [comment]
Sounds like a good plan to me.

**Comment by unioncapitalboston** [comment]
Duncan - this is immensely helpful research - much thanks! It sounds like a great plan and I totally trust your call on this. We can cover the costs whatever they end up being.

Another way to decrease the photo storage size is to have them wiped clean a set amount of time after approval. I'm totally cool with images being deleted 2 weeks after approval. That gives Partner Organizations time to see and download the images if they want, but otherwise there is no reason to keep them.

If that isn't a heavy lift, will that help reduce the cost when switching too? Either way is cool.

Much thanks!

Eric

**Comment by drenfr01** [comment]
we are now on compose and s3. woot woot!
